src/primaite/agents/git/git_policy.py

```diff
@@ -1,28 +1,36 @@
 import torch
 import torch.nn.functional as F
 from torch import nn
-from torch_geometric.nn import GATConv
-from torch_geometric.nn.pool import global_max_pool
+from torch_geometric.nn import GATConv, LayerNorm, global_add_pool
 from torch.optim import Adam
 from transformers import BertModel, BertTokenizer, BitsAndBytesConfig
 from transformers import AutoModelForCausalLM, AutoTokenizer
 from transformers.models.llama.modeling_llama import LlamaForCausalLM
 from typing import List, Tuple, Dict
 import numpy as np
+import logging
+
+
+logging.getLogger().setLevel(logging.INFO)
 
 
 class GraphEmbedding(nn.Module):
-    def __init__(self, in_channels, output_dim, hidden_dim, device: str = "cuda:1"):
+    def __init__(self, in_channels, output_dim, hidden_dim, n_tokens: int = 10, device: str = "cuda:1"):
         super().__init__()
         self.gat = GATConv(in_channels=in_channels, out_channels=hidden_dim)
-        self.linear = nn.Linear(in_features=hidden_dim, out_features=output_dim)
+        self.layer_norm = LayerNorm(hidden_dim)
+        self.linear = nn.Linear(in_features=hidden_dim, out_features=n_tokens * output_dim)
         self.device = device
+        self.n_tokens = n_tokens
+        self.output_dim = output_dim
 
     def forward(self, x, edge_index):
+        n_nodes = x.shape[0]
         x = self.gat(x.to(self.device), edge_index.to(self.device))
+        x = global_add_pool(self.layer_norm(x), torch.LongTensor([0 for _ in range(n_nodes)]).to(self.device))
```

Comment:
do you need to pass the batch here? also i think it's quite slow to use python list comprehensions here to generate that vector of 0's
better use torch.zeros(n_nodes)

Reply:
in my defence I inhezzad it from jacks

Reply:
done